packetbeat/beater/packetbeat.go

```diff
@@ -172,9 +170,9 @@ func (pb *packetbeat) Run(b *beat.Beat) error {
 		service.Stop()
 	}
 
-	waitShutdown := pb.cmdLineArgs.waitShutdown
-	if waitShutdown != nil && *waitShutdown > 0 {
-		time.Sleep(time.Duration(*waitShutdown) * time.Second)
+	timeout := b.Config.ShutdownTimeout
```

Comment:
The problem is that packetbeat does not have the same logic here as filebeat. On the filebeat side it waits for all events to be published and then stops. If all events are sent before `shutdown_timeout` is reached, it will directly shut down. Because this tracking if all events are sent does not exist in packetbeat, packetbeat just "sleeps" for the time. 

The reason I worry about this it because there are two different behaviours with 1 config option. As this problem can't be covered in this PR I think we should solve it through documentation.

Also this code should probably not be in packetbeat anymore, but on the beats side instead. Otherwise each beat has to implement it.